charms/mariadb/reactive/mariadb.py

```diff
@@ -1,14 +1,29 @@
 import pymysql
 from charmhelpers.core import hookenv
 from charms import layer
-from charms.reactive import set_flag, when, when_not, clear_flag
+from charms.reactive import set_flag, when, when_not, clear_flag, endpoint_from_name
 
 
 @when('charm.mariadb.started')
 def charm_ready():
     layer.status.active('')
 
 
+@when('charm.mariadb.started')
+def configure_mysql():
+    mysql = endpoint_from_name('mysql')
+
+    for i in range(len(mysql.relations)):
+        mysql.provide_database(
+            request_id=i,
+            database_name=hookenv.config('database'),
+            port=hookenv.config('port'),
+            host=hookenv.application_name(),
+            user='root',
+            password=hookenv.config('root-password'),
```

Comment:
The IaaS implementation of this charm generates a separate database and user per related application so that it can support multiple clients.  Maybe supporting multiple clients isn't as expected in the k8s world (or maybe even running a DB in-cluster at all), though I would expect it to get there eventually.  This is fine for a first pass, but we'll probably want to round the functionality here out at some point.

Reply:
Sure, that sounds good. Yes, we can expect multiple clients in general (and the semi-official recommendation is that you shouldn't run stateful services in kubernetes, but people still do and things like statefulsets are giving kubernetes better support for it), but I wanted to start simple for now.